src/x/debugdump/debug.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package debugdump
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+)
+
+// DataProvider returns an instance of a provider for debug information
+type DataProvider interface {
```

Comment:
The name of this package is already so well scoped that I wonder if this could just be called provider.

I.E

`debugdump.Provider` vs. `debugdump.DataProvider`

That feels a little more intuitive to me because then it seems more obvious that a provider is providing debug information as opposed to jus "data" but I don't have super strong feelings

Reply:
Alternatively:

```
type Source interface {
    Write(w io.Writer) error
}
```

Seems reasonable to me too